pkg/destroy/aws/aws.go

```diff
@@ -160,132 +157,162 @@ func (o *ClusterUninstaller) RunWithContext(ctx context.Context) error {
 		logger:  o.Logger,
 	}
 
-	deleted, err := terminateEC2InstancesByTags(ctx, ec2.New(awsSession), iamClient, o.Filters, o.Logger)
+	deleted := destroyedResources{}
+
+	resourceChanges, err := terminateEC2InstancesByTags(ctx, ec2.New(awsSession), iamClient, o.Filters, o.Logger)
+	deleted.merge(resourceChanges)
 	if err != nil {
-		return err
+		// Collect the ARNs of all of the tagged resources which have been deleted.
+		// Use a context without a deadline since the context supplied has already expired.
+		if _, err := o.forAllTaggedResources(
+			context.Background(),
+			tagClients,
+			iamRoleSearch,
+			iamUserSearch,
+			func(arnString string, filter Filter) (ignored bool, returnErr error) {
+				if deleted[arnString] {
+					return true, nil
+				}
+				deleted[arnString] = false
+				return false, nil
+			},
+		); err != nil {
+			o.Logger.WithError(err).Error("failed to get tagged resources")
+		}
+		return deleted.extractRemaining(), err
 	}
 
+	notDeleted := deleted.extractRemaining()
```

Comment:
This is making it really difficult to follow the logic, deleted - notDeleted being extracted from same deleted.
Also https://github.com/openshift/installer/pull/3772/files#diff-ab65f0e2ba0237c5cc1429673c22cc08R207-R208

Suffers from the same complexity, we are merging resource changes to deleted but then we also record to deleted ??

I think we need to simplify the tracking here a little bit.